spec/requests/admin/badges_controller_spec.rb

```diff
@@ -244,6 +260,114 @@
 
         expect(response.status).to eq(422)
       end
+
+      context "when grant_existing_holders is true" do
```

Comment:
I feel like alot of the test cases here should be moved under the `BadgeGranter` service object spec. For controller tests, I usually prefer to test the various response code that we expect. Success, errors etc.

Reply:
I've moved most of the logic in `Admin::BadgesController#mass_award` into a new static method in `BadgeGranter` and moved some of the tests here to `BadgeGranter` spec file. Let me know if I should move more.